src/spath.c

```diff
@@ -905,8 +911,24 @@ int spath_slice(spath* path, int offset, int length)
     return SPATH_SUCCESS;
   }
 
-  /* force offset into range */
   int components = path->components;
+  offset = (offset >= 0) ? offset : (offset + components);
+  if (offset < 0 || offset > components) {
+    fprintf(stderr, "starting index not in range of components @ %s:%d",
+      __FILE__, __LINE__
+    );
+    return SPATH_FAILURE;
+  }
+  
+  if (length >= 0 && (offset + length) > components) {
+    fprintf(stderr, "(offset + length) > components @ %s:%d",
+      __FILE__, __LINE__
+    );
+    return SPATH_FAILURE;
+  }
```

Comment:
This new code will make it invalid to use offsets and lengths combinations that correspond to values that go beyond the range of actual components.  It changes the semantics of the functions in a way that may break existing code (like SCR).  If we decide to keep this change, we should proactively check and fix incorrect usage in SCR, since SCR likely does not not check spath return codes.

Reply:
Hi Adam. I am not sure about the origin of this code. I did not put it in and it was not a part of any merge conflicts tha tI had to resolve. Should I take it out?

Reply:
Adam, what's your word on this?

Reply:
@kosinovsky , I haven't had a chance to dig into this.  My gut reaction is that we should avoid adding this code.  However, I vaguely remember us discussing this before.  We might have settled on leaving it in then, but I forget reason.  Was hoping to review the code / email history to investigate.